Fix ensure_single_hinge()

smith120bh opened the pull request:
Fixes #36

ritchie46 commented on anastruct/fem/postprocess.py:
```diff
@@ -93,24 +93,31 @@ def __init__(self, system: "SystemElements"):
         self.system = system
 
     @staticmethod
```
@staticmethod should be removed if `self ` is used.

smith120bh replied:
Fixed

ritchie46 commented:
The changes lead to failure in 4 tests. See the logs of https://travis-ci.org/github/ritchie46/anaStruct/builds/712519354

smith120bh commented:
Now all passed, and I now know where that test script is before I PR anything in the future :-)

ritchie46 commented on anastruct/fem/system_components/util.py:
```diff
@@ -21,35 +21,65 @@ def ensure_single_hinge(
     node_id2
         Id of the node in the system
     """
-    if spring is not None and 0 in spring:
```
Good!

ritchie46 commented on anastruct/fem/system_components/util.py:
```diff
@@ -21,35 +21,65 @@ def ensure_single_hinge(
+                        system.node_map[node_id].hinge = True
+                        del spring[node_nr]  # too many hinges at that element.
+
+                elif node_nr not in spring.keys() or (
```
Is this not always True if previous statement evaluates to False? Could we replace this with else?

smith120bh replied:
Yes it is... This is why reviews are always helpful :-)

smith120bh commented:
Hi Ritchie - I use black myself for most of my work, so my edits there were already autoformatted by it. I've just fixed the spot you noted could be made a simpler `else` statement, and added the test case.
